modules/core/include/opencv2/core/private.hpp

```diff
@@ -230,44 +294,193 @@ static inline IppiPoint ippiPoint(int x, int y)
+        inMemBorder.borderBottom = (Ipp32u)(origSize.height - src.rows - offset.y);
+    }
+
+    dst.Init(ippiSize(src.size()), ippiGetDataType(src.depth()), src.channels(), inMemBorder, (void*)src.ptr(), src.step);
+}
+
+static inline ::ipp::IwiImage ippiGetImage(const cv::Mat &src)
+{
+    ::ipp::IwiImage image;
+    ippiGetImage(src, image);
+    return image;
+}
+
+static inline IppiBorderType ippiGetBorder(::ipp::IwiImage &image, int ocvBorderType, IppiBorderSize &borderSize)
+{
+    int            inMemFlags = 0;
+    IppiBorderType border     = ippiGetBorderType(ocvBorderType & ~cv::BORDER_ISOLATED);
+    if((int)border == -1)
+        return (IppiBorderType)0;
+
+    if(!(ocvBorderType & cv::BORDER_ISOLATED))
+    {
+        if(image.m_inMemSize.borderLeft)
+        {
+            if(image.m_inMemSize.borderLeft >= borderSize.borderLeft)
+                inMemFlags |= ippBorderInMemLeft;
+            else
+                return (IppiBorderType)0;
+        }
+        else
+            borderSize.borderLeft = 0;
+        if(image.m_inMemSize.borderTop)
+        {
+            if(image.m_inMemSize.borderTop >= borderSize.borderTop)
+                inMemFlags |= ippBorderInMemTop;
+            else
+                return (IppiBorderType)0;
+        }
+        else
+            borderSize.borderTop = 0;
+        if(image.m_inMemSize.borderRight)
+        {
+            if(image.m_inMemSize.borderRight >= borderSize.borderRight)
+                inMemFlags |= ippBorderInMemRight;
+            else
+                return (IppiBorderType)0;
+        }
+        else
+            borderSize.borderRight = 0;
+        if(image.m_inMemSize.borderBottom)
+        {
+            if(image.m_inMemSize.borderBottom >= borderSize.borderBottom)
+                inMemFlags |= ippBorderInMemBottom;
+            else
+                return (IppiBorderType)0;
+        }
+        else
+            borderSize.borderBottom = 0;
+    }
+    else
+        borderSize.borderLeft = borderSize.borderRight = borderSize.borderTop = borderSize.borderBottom = 0;
+
+    return (IppiBorderType)(border|inMemFlags);
+}
+
+static inline ::ipp::IwValue ippiGetValue(const cv::Scalar &scalar)
+{
+    return ::ipp::IwValue(scalar[0], scalar[1], scalar[2], scalar[3]);
+}
+
+static inline int ippiSuggestThreadsNum(const ::ipp::IwiImage &image, double multiplier)
+{
+    int threads = cv::getNumThreads();
+    if(image.m_size.height > threads)
+    {
+        size_t opMemory = (int)(image.m_typeSize*image.m_channels*image.m_size.width*image.m_size.height*multiplier);
+        int l2cache  = 0;
+        ippGetL2CacheSize(&l2cache);
+        if(!l2cache)
+            l2cache = 100000;
+
+        return IPP_MAX(1, (IPP_MIN((int)(opMemory/(l2cache*0.6)), threads)));
+    }
+    return 1;
+}
+#endif
+
+static inline int ippiSuggestThreadsNum(const cv::Mat &image, double multiplier)
+{
+    int threads = cv::getNumberOfCPUs();
```

Comment:
Probably this should be consistent with implementation above: `cv::getNumThreads()` vs `cv::getNumberOfCPUs()`?